trace-gradle-plugin/src/functionalTest/java/io/bitrise/trace/plugin/util/FunctionalTestHelper.java

```diff
@@ -0,0 +1,298 @@
+    public void publishTraceGradlePlugin() {
+        logger.lifecycle("Publishing plugin, to make sure it is up-to-date");
+        FunctionalTestUtils.publishTraceGradlePlugin();
+        logger.lifecycle("Publishing plugin finished");
+    }
+
+    /**
+     * Prints multiple lines with the given text in a pretty ASCII text box.
+     *
+     * @param text the text to print.
+     */
+    public void logInAsciiBox(@NonNull final String text) {
+        final int textLength = text.length();
+        final int paddingLength = 30;
+        final int lineLength = paddingLength + textLength;
+
+        final StringBuilder stringBuilder = new StringBuilder();
+
+        stringBuilder.append("\n");
+        stringBuilder.append(getTestNameBorderLogLine(lineLength));
+        stringBuilder.append(getTestNameEmptyLogLine(lineLength));
+        stringBuilder.append(getTestNameLogLine(text, paddingLength));
+        stringBuilder.append(getTestNameEmptyLogLine(lineLength));
+        stringBuilder.append(getTestNameBorderLogLine(lineLength));
+
+        logger.lifecycle(stringBuilder.toString());
+    }
+
+    /**
+     * Prints multiple lines with the name of the test in a pretty ASCII text box.
+     *
+     * @param testName the name of the test to print.
+     */
+    public void logTestNameInAsciiBox(@NonNull final TestName testName) {
+        logInAsciiBox(testName.getMethodName());
+    }
+
+    /**
+     * Prints the top or bottom line. Contains separators only.
+     *
+     * @param lineLength the length of the lines.
+     * @return the line.
+     */
+    @NonNull
+    private String getTestNameBorderLogLine(final int lineLength) {
+        final StringBuilder stringBuilder = new StringBuilder();
+        stringBuilder.append("+");
+        stringBuilder.append(concatWord("=", lineLength));
+        stringBuilder.append("+");
+        stringBuilder.append("\n");
+        return stringBuilder.toString();
+    }
+
+    /**
+     * Prints a line with empty content.
+     *
+     * @param lineLength the length of the lines.
+     * @return the line.
+     */
+    @NonNull
+    private String getTestNameEmptyLogLine(final int lineLength) {
+        final StringBuilder stringBuilder = new StringBuilder();
+        stringBuilder.append("|");
+        stringBuilder.append(concatWord(" ", lineLength));
+        stringBuilder.append("|");
+        stringBuilder.append("\n");
+        return stringBuilder.toString();
+    }
+
+    /**
+     * Prints the line with the name of the test.
+     *
+     * @param testName      the given text to display.
+     * @param paddingLength the length of padding for the line, excluding starting and ending cahars.
+     * @return the line.
+     */
+    @NonNull
+    private String getTestNameLogLine(final String testName, final int paddingLength) {
+        final StringBuilder stringBuilder = new StringBuilder();
+        stringBuilder.append("|");
+        stringBuilder.append(concatWord(" ", paddingLength / 2));
```

Comment:
nit: what happens if padding length here isn't divisible by 2? I know that we're setting it up earlier to be 30, but as anyone could use this method - it might be worth just having a Math.floor here too?

Reply:
Good catch! The method concatWord expect an int as the second param. This means non divisible numbers will be rounded down. This will result in uneven line lengths (empty lines will be 1 char longer). There are different strategies to mitigate this:
1. Use only even numbers as padding, so there will be no such case. We can update the logic, to make the padding even, if it is not even.
2. Enable non even padding, this will result in asymmetric position of the test name

I would go with the first, because it is better looking. For that we can:
- indicate that use only even number as input, otherwise expect uneven box lines
- convert every input to even number (if they are not already even)
Please let me know your thoughts on this.

Reply:
I think that sounds like a good strategy :) Perhaps we can make the current padding a constant at the top of the page with a comment with why it has to be an even number?

Reply:
Making it constant would make sense if more things would have used it, so I updated the javadoc of the given method to reflect on this.